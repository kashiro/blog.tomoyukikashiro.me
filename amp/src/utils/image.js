export const headerBgUrl = (day) => {
  return day ? `/images/${ day % 7 }.jpg` : `/images/home.jpg`
}

export const headerBgClass = (day) => {
  return `header__bg_${ day % 7 }`
}
