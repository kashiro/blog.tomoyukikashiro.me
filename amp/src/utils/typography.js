import Typography from 'typography'
import typographyThemeKirkham from 'typography-theme-kirkham'

const typography = new Typography(typographyThemeKirkham)

// Hot reload typography in development.
if (process.env.NODE_ENV !== 'production') {
  typography.injectStyles()
}

export default typography
export const rhythm = typography.rhythm
export const scale = typography.scale
