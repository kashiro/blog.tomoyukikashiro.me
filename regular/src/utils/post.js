// import moment from 'moment'

export default class Post {

  static LANGS = {
    JA: 'ja',
    EN: 'en-US'
  }

  constructor(node) {
    this.node = node
  }
  
  get key() {
    return `${this.lang}-${this.slug}`
  }

  get title() {
    return this.node.frontmatter.title.trim()
  }

  get date() {
    if (!this.node.frontmatter.date) return
    return new Date(this.node.frontmatter.date.trim())
  }
  
  get type() {
    return 'article'
  }
  
  get isoDate() {
    return this.date.toISOString()
  }
  
  get formatDate() {
    return this.date.toLocaleString('en-US', { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' })
    // return moment(this.date.toISOString()).format('dddd LL')
  }
  
  get summary() {
    return this.node.frontmatter.summary ? this.node.frontmatter.summary.trim() : ''
  }
  
  get slug() {
    return this.node.frontmatter.slug.trim()
  }
  
  get lang() {
    return this.node.frontmatter.lang.trim()
  }
  
  get isEn() {
    return this.lang === Post.LANGS.EN
  }
  
  get isJa() {
    return this.lang === Post.LANGS.JA
  }
  
  get alternativeLang() {
    return this.isEn ? Post.LANGS.JA : Post.LANGS.EN
  }
  
  get tags() {
    if (this.node.frontmatter.tags) {
      return this.node.frontmatter.tags.map(tag => tag.trim().toLowerCase()) 
    } else {
      return []
    }
  }
  
  get hasTags() {
    return !!this.tags.length
  }

  path(lang) {
    const langPath = (lang || this.lang) === Post.LANGS.JA ? '/ja/' : '/'
    return `/post${langPath}${this.slug}`
  }
  
  get alternativeLangPath() {
    return this.path(this.alternativeLang)
  }
  
  get html() {
    return this.node.html
  }
  
  get canonicalUrl() {
    return this.node.frontmatter.canonicalUrl
  }
}
  

