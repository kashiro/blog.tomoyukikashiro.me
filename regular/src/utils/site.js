import Post from './post'

export default class Site {

  constructor(siteMetadata) {
    this.meta = siteMetadata
  }
  
  // --- title ---
  
  get title() {
    return this.meta.title
  }
  
  get tagsPageTitle() {
    return `TAGS | ${this.title}`
  }
  
  tagPagePrefix(tag, totalCount) {
    return `Post${totalCount === 1 ? "" : "s"} tagged with "${tag.toUpperCase()}"`
  }
  
  tagPageTitle(tag, totalCount) {
    return `${this.tagPagePrefix(tag, totalCount)} TAG | ${this.title}`
  }

  // --- description ---
  get description() {
    return this.meta.description
  }
  
  get tagsPageDescription() {
    return `TAGS | ${this.meta.description}`
  }

  tagPageDescription(tag, totalCount) {
    return `${this.tagPagePrefix(tag, totalCount)} TAG | ${this.description}`
  }
  
  postPageDescription(post) {
    return post.summary
  }
  
  
  // --- url ---
  
  get url() {
    return this.meta.siteUrl
  }
  
  get ampUrl() {
    return this.meta.ampUrl
  }
  
  postUrl(post, lang) {
    return `${this.url}${post.path(lang)}/`
  }
  
  postEnUrl(post) {
    return this.postUrl(post, Post.LANGS.EN)
  }
  
  postAlternativeLangUrl(post) {
    return this.postUrl(post, post.alternativeLang)
  }
  
  canonicalPostUrl(post, lang) {
    if (post.canonicalUrl) return post.canonicalUrl
    return this.postUrl(post, lang)
  }
  
  canonicalPostAmpUrl(post) {
    return `${this.ampUrl}${post.path()}/`
  }

  get tagsPageUrl() {
    return `${this.url}/tags`
  }
  
  tagPageUrl(tag) {
    return `${this.url}/tag/${tag.toLowerCase()}/`
  }
  
  get profileUrl() {
    return this.meta.profileUrl
  }
  
  // --- others ---

  get type() {
    return 'website'
  }

  get author() {
    return this.meta.author
  }
  
  get twitterUserName() {
    return this.meta.twitterUserName
  }
}
