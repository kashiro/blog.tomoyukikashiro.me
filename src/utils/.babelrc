{
  "presets": [
    [
      "@babel/preset-env",
      {
        "loose": true,
        "modules": "commonjs",
        "useBuiltIns": "usage",
        "targets": {
          "node": true
        }
      }
    ]
  ],
  "plugins": [
    [
      "@babel/plugin-proposal-class-properties",
      {
        "loose": true
      }
    ]
  ]
}
