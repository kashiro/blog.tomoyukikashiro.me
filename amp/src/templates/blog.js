import React from 'react'
import Helmet from 'react-helmet'
import { Link,graphql } from 'gatsby'

import Layout from '../components/Layout'
import { headerBgClass } from '../utils/image'
import ArticleBreadCrumb from '../components/ld_json/ArticleBreadCrumb'
import Article from '../components/ld_json/Article'
import Header from '../components/Header'
import moment from 'moment'

import HeaderStyles from '../components/Header.module.css'
import LabelSvg from '../assets/images/label.svg'

export const BlogPostTemplate = ({author, date, title, slug, tags, html, content}) => {
  const _date = new Date(date)
  return (
    <article>
      <Header klass={headerBgClass(_date.getDate())} text={title} link={`/post/${ slug }/`}>
        <div className={ HeaderStyles.header__meta }>
          <address className={ `${ HeaderStyles.header__author } text-elegant` }>By { author }</address>
          <time className={ `${ HeaderStyles.header__publish_date } text-elegant` } dateTime={ _date.toISOString() }> on { moment(_date.toISOString()).format('dddd LL')  }</time>
        </div>
        { tags
          ? <ul className={ `${ HeaderStyles.header__tags } clearfix` }>
            {
              tags.map(tag => (
                <li className={ `${ HeaderStyles.header__tag } tag` } key={ tag.toLowerCase() }>
                  <i><LabelSvg className={HeaderStyles.header__title_icon} /></i>
                  <Link to={ `/tag/${ tag.toLowerCase() }` }>{ tag.toUpperCase() }</Link>
                </li>
              ))
            }
          </ul>
          : null
        }
      </Header>
      { content
        ? <div className="markdown-body body">{ content }</div>
        : <div className="markdown-body body" dangerouslySetInnerHTML={{ __html: html }}></div>
      }
    </article>
  )
}

class BlogPost extends React.Component {
  render() {
    const post = this.props.data.markdownRemark
    const siteMeatadata = this.props.data.site.siteMetadata
    
    return (
      <Layout>
        <Helmet>
          <title>{ post.frontmatter.title }</title>
          <meta name="description" content={ post.frontmatter.summary || siteMeatadata.description } />
          <link rel="canonical" href={ `${siteMeatadata.siteUrl}/post/${post.frontmatter.slug}/` } />
        </Helmet>
        <ArticleBreadCrumb post={post.frontmatter} />
        <Article post={post.frontmatter} />
        <BlogPostTemplate
          author={siteMeatadata.author}
          date={post.frontmatter.date}
          title={post.frontmatter.title}
          slug={post.frontmatter.slug}
          tags={post.frontmatter.tags}
          html={post.html} />
      </Layout>
    )
  }
}

export default BlogPost

export const pageQuery = graphql`
  query ($slug: String!) {
    site {
      siteMetadata {
        siteUrl
        author
        description
      }
    }
    markdownRemark(frontmatter: { slug: { eq: $slug } }) {
      id
      excerpt
      html
      frontmatter {
        title
        slug
        summary
        tags
        date(formatString: "MMMM DD, YYYY")
      }
    }
  }
`

