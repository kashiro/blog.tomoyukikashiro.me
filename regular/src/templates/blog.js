import React from 'react'
import Helmet from 'react-helmet'
import { Link,graphql } from 'gatsby'

import Layout from '../components/Layout'
import MetaSocial from '../components/MetaSocial'
import { headerBgUrl, headerBgClass } from '../utils/image'
import ArticleBreadCrumb from '../components/ld_json/ArticleBreadCrumb'
import Article from '../components/ld_json/Article'
import Header from '../components/Header'
import moment from 'moment'
import Disqus from '../components/Disqus'

import HeaderStyles from '../components/Header.module.css'
import LabelSvg from '../assets/images/label.svg'

export class BlogPostTemplate extends React.Component {
  state = {
    ShareButtonContainer: null,
    ShareButton: null
  }
  
  componentDidMount() {
    if (!navigator.share) return
    import(/* webpackChunkName: "sharebutton" */ '../components/ShareButton')
      .then(module => {
        this.setState({
          ShareButtonContainer: module.ShareButtonContainer,
          ShareButton: module.ShareButton
        })
      })
  }

  render () {
    const {author, date, title, slug, tags, html, content, url} = this.props
    const {ShareButtonContainer, ShareButton} = this.state
    const _date = new Date(date)
    return (
      <article>
        <Header klass={headerBgClass(_date.getDate())} text={title} link={`/post/${ slug }/`}>
          <div className={ HeaderStyles.header__meta }>
            <address className={ `${ HeaderStyles.header__author } text-elegant` }>By { author }</address>
            <time className={ `${ HeaderStyles.header__publish_date } text-elegant` } dateTime={ _date.toISOString() }> on { moment(_date.toISOString()).format('dddd LL')  }</time>
          </div>
          { tags
            ? <ul className={ `${ HeaderStyles.header__tags } clearfix` }>
              {
                tags.map(tag => (
                  <li className={ `${ HeaderStyles.header__tag } tag` } key={ tag.toLowerCase() }>
                    <i><LabelSvg className={HeaderStyles.header__title_icon} /></i>
                    <Link to={ `/tag/${ tag.toLowerCase() }` }>{ tag.toUpperCase() }</Link>
                  </li>
                ))
              }
            </ul>
            : null
          }
        </Header>
        <div className="markdown-body body">
          { content
            ? <div>{ content }</div>
            : <div dangerouslySetInnerHTML={{ __html: html }}></div>
          }
          { ShareButtonContainer && ShareButton
            ? <ShareButtonContainer><ShareButton url={url} title={title}/></ShareButtonContainer>
            : null
          }
        </div>
      </article>
    )
  }
}

class BlogPost extends React.Component {
  render() {
    const post = this.props.data.markdownRemark
    const siteMeatadata = this.props.data.site.siteMetadata
    const canonicalUrl = `${siteMeatadata.siteUrl}/post/${post.frontmatter.slug}/`
    
    return (
      <Layout>
        <Helmet>
          <title>{ post.frontmatter.title }</title>
          <meta name="description" content={ post.frontmatter.summary || siteMeatadata.description } />
          <link rel="canonical" href={ canonicalUrl } />
          <link rel="amphtml" href={ `${siteMeatadata.ampUrl}/post/${post.frontmatter.slug}/` } />
        </Helmet>
        <MetaSocial
          title={post.frontmatter.title }
          description={ post.frontmatter.summary || siteMeatadata.description }
          type="article"
          url={ `${siteMeatadata.siteUrl}/post/${post.frontmatter.slug}/` }
          image={ headerBgUrl(new Date(post.frontmatter.date).getDate()) }
          tags={ post.frontmatter.tags }
          published={ new Date(post.frontmatter.date) }
        />
        <ArticleBreadCrumb post={post.frontmatter} />
        <Article post={post.frontmatter} />
        <BlogPostTemplate
          author={siteMeatadata.author}
          date={post.frontmatter.date}
          title={post.frontmatter.title}
          slug={post.frontmatter.slug}
          tags={post.frontmatter.tags}
          url={canonicalUrl}
          html={post.html} />
        <aside className="body">
          <Disqus
            siteName={siteMeatadata.disqusSiteName}
            siteUrl={siteMeatadata.siteUrl}
            path={`post/${post.frontmatter.slug}`}></Disqus>
        </aside>
      </Layout>
    )
  }
}

export default BlogPost

export const pageQuery = graphql`
  query ($slug: String!) {
    site {
      siteMetadata {
        siteUrl
        ampUrl
        author
        description
        disqusSiteName
      }
    }
    markdownRemark(frontmatter: { slug: { eq: $slug } }) {
      id
      excerpt
      html
      frontmatter {
        title
        slug
        summary
        tags
        date(formatString: "MMMM DD, YYYY")
      }
    }
  }
`

