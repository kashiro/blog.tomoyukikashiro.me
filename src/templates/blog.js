import React from 'react'
import Helmet from 'react-helmet'
import { graphql } from 'gatsby'

import Layout from '../components/Layout'
import MetaSocial from '../components/MetaSocial'
import { headerBgUrl, headerBgClass } from '../utils/image'
import ArticleBreadCrumb from '../components/ld_json/ArticleBreadCrumb'
import Article from '../components/ld_json/Article'
import Header from '../components/Header'
import FooterStyle from './BlogFooter.module.css'

import HeaderStyles from '../components/Header.module.css'
import Site from '../utils/site'
import Post from '../utils/post'

export const BlogPostHead = ({ post, site, hasAlternate = false}) => (
  <React.Fragment>
    <Helmet>
      <title>{ post.title }</title>
      <meta name="description" content={ site.postPageDescription(post) } />
      <link rel="amphtml" href={ site.postUrl(post) } />
      <link rel="canonical" href={ site.postUrl(post) } />
      <link rel="alternate" href={ site.postUrl(post) } hrefLang={ post.lang } />
      { hasAlternate ? <link rel="alternate" href={ site.postAlternativeLangUrl(post) } hrefLang={ post.alternativeLang } />: ''}
    </Helmet>
    <MetaSocial
      site={ site }
      title={ post.title }
      description={ post.summary || site.description }
      type={ post.type }
      url={ site.postUrl(post) }
      image={ headerBgUrl(post.date.getDate()) }
      published={ post.isoDate }
      lang={ post.lang }
    />
    <ArticleBreadCrumb post={ post } site={site} />
    <Article post={ post } site={site} />
  </React.Fragment>
)

export const BlogPostFooter = ({ url, twitterUserName }) => {
  const tweet = encodeURIComponent(`.@${twitterUserName} Feel free to ask question :)  `)
  const tweetUrl = `https://twitter.com/intent/tweet?text=${tweet}&url=${encodeURIComponent(url)}` 
  return (
    <div className={FooterStyle.l_blog_footer}>This blog doesn't have comment function <br/> so feel free to contact me via <a href={tweetUrl} target="_blank">Twitter</a></div>
  )
}

export class BlogPostTemplate extends React.Component {
  render () {
    const post = new Post({frontmatter: this.props, html: this.props.html})
    return (
      <article>
        <Header klass={headerBgClass(post.date.getDate())} text={ post.title } link={ `${post.path()}/` }>
          <div className={ HeaderStyles.header__meta }>
            <address className={ `${ HeaderStyles.header__author } text-elegant` }>By { this.props.author }</address>
            { post.isoDate
              ? <time className={ `${ HeaderStyles.header__publish_date } text-elegant` } dateTime={ post.isoDate }> on { post.formatDate }</time>
              : null
            }
          </div>
        </Header>
        <div className="markdown-body body">
          { this.props.content
            ? <div>{ this.props.content }</div>
            : <div dangerouslySetInnerHTML={{ __html: post.html }}></div>
          }
        </div>
      </article>
    )
  }
}

class BlogPost extends React.Component {
  render() {
    const post = new Post(this.props.data.markdownRemark)
    const site = new Site(this.props.data.site.siteMetadata)
    const hasAlternate = this.props.pageContext.hasAlternate || false
    
    return (
      <Layout site={site}>
        <BlogPostHead site={site} post={post} hasAlternate={hasAlternate} />
        <BlogPostTemplate
          author={ site.author }
          date={ `${post.date}` }
          title={ post.title }
          slug={ post.slug }
          lang={ post.lang }
          url={ site.postUrl(post) }
          html={ post.html } />
        <BlogPostFooter url={site.postEnUrl(post)} twitterUserName={site.twitterUserName}/>
      </Layout>
    )
  }
}

export default BlogPost

export const pageQuery = graphql`
  query ($slug: String!, $lang: String!) {
    site {
      siteMetadata {
        title
        author
        description
        siteUrl
        profileUrl
        twitterUserName
        ampUrl
      }
    }
    markdownRemark(frontmatter: { slug: { eq: $slug }, lang: { eq: $lang } }) {
      id
      excerpt
      html
      frontmatter {
        title
        slug
        summary
        lang
        date(formatString: "MMMM DD, YYYY")
      }
    }
  }
`

