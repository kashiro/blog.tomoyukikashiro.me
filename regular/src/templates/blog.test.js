import React from 'react'
import renderer from 'react-test-renderer'
import { requestAnimationFrame } from '../../__tests__/utils'
import { BlogPostHead } from './blog'

describe('BlogPostHead', function () {
  const site = {
    postPageDescription: () => ('post description'),
    canonicalPostAmpUrl: () => ('https://amp.example.com'),
    postUrl: () => ('https://example.com'),
    canonicalPostUrl: () => ('https://canonical.example.com'),
    title: 'test site title',
    url: 'https://example.com',
    author: 'test author',
    twitterUserName: 'testAccount',
  }
  const post = {
    title: 'test post title',
    summary: 'test post summary',
    type: 'article',
    date: new Date('2018-12-24'),
    isoDate: new Date('2018-12-24').toISOString(),
    lang: 'en-US',
    tags: ['test1', 'test2'],
    path: () => ('/post/test-slug')
  }
  let headElement;
  beforeEach(() => {
    headElement = headElement || document.head || document.querySelector("head");

    // resets DOM after each run
    headElement.innerHTML = "";
  })
  describe('When hasAlternate is false', function () {
    it('BlogPostHead render correctly', function (done) {
      renderer.create(<BlogPostHead site={ site } post={post} hasAlternate={false}/>).toJSON()
      requestAnimationFrame(() => {
        const tree = renderer.create(headElement.innerHTML).toJSON()
        expect(tree).toMatchSnapshot()
        done()
      })
    })
  })
  describe('When hasAlternate is true', function () {
    site.postAlternativeLangUrl = () => ('https://example.com/post/ja/test-slug/')
    post.alternativeLang = 'ja'
    it('BlogPostHead render correctly', function (done) {
      renderer.create(<BlogPostHead site={ site } post={post} hasAlternate={true}/>).toJSON()
      requestAnimationFrame(() => {
        const tree = renderer.create(headElement.innerHTML).toJSON()
        expect(tree).toMatchSnapshot()
        done()
      })
    })
  })
})
