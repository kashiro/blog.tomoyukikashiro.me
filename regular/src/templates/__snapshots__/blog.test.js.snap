// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostHead When hasAlternate is false BlogPostHead render correctly 1`] = `
"<title>test post title</title><link rel=\\"amphtml\\" href=\\"https://amp.example.com\\" data-react-helmet=\\"true\\"><link rel=\\"canonical\\" href=\\"https://example.com\\" data-react-helmet=\\"true\\"><link rel=\\"alternate\\" href=\\"https://example.com\\" hreflang=\\"en\\" data-react-helmet=\\"true\\"><meta name=\\"description\\" content=\\"post description\\" data-react-helmet=\\"true\\"><meta property=\\"og:title\\" content=\\"test post title\\" data-react-helmet=\\"true\\"><meta property=\\"og:type\\" content=\\"article\\" data-react-helmet=\\"true\\"><meta property=\\"og:image\\" content=\\"https://example.com/images/3.jpg\\" data-react-helmet=\\"true\\"><meta property=\\"og:url\\" content=\\"https://example.com\\" data-react-helmet=\\"true\\"><meta property=\\"og:description\\" content=\\"test post summary\\" data-react-helmet=\\"true\\"><meta property=\\"og:locale\\" content=\\"en\\" data-react-helmet=\\"true\\"><meta property=\\"og:site_name\\" content=\\"test site title\\" data-react-helmet=\\"true\\"><meta property=\\"article:published_time\\" content=\\"2018-12-24T00:00:00.000Z\\" data-react-helmet=\\"true\\"><meta property=\\"profile:username\\" content=\\"test author\\" data-react-helmet=\\"true\\"><meta property=\\"article:tag\\" content=\\"test1\\" data-react-helmet=\\"true\\"><meta property=\\"article:tag\\" content=\\"test2\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:card\\" content=\\"summary_large_image\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:creator\\" content=\\"@testAccount\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:title\\" content=\\"test post title\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:description\\" content=\\"test post summary\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:image\\" content=\\"https://example.com/images/3.jpg\\" data-react-helmet=\\"true\\"><script type=\\"application/ld+json\\" data-react-helmet=\\"true\\">{
        \\"@context\\": \\"http://schema.org\\",
        \\"@type\\": \\"BreadcrumbList\\",
        \\"itemListElement\\": [
          {
            \\"@type\\": \\"ListItem\\",
            \\"position\\": 1,
            \\"item\\": {
              \\"@id\\": \\"https://example.com/\\",
              \\"name\\": \\"HOME\\",
              \\"image\\": \\"https://example.com/images/home.jpg\\"
            }
          },
          {
            \\"@type\\": \\"ListItem\\",
            \\"position\\": 2,
            \\"item\\": {
              \\"@id\\": \\"https://example.com/post/test-slug/\\",
              \\"name\\": \\"test post title\\",
              \\"image\\": \\"https://example.com/images/3.jpg\\"
            }
          }
        ]
      }</script><script type=\\"application/ld+json\\" data-react-helmet=\\"true\\">{
          \\"@context\\": \\"http://schema.org\\",
          \\"@type\\": \\"Article\\",
          \\"mainEntityOfPage\\": {
            \\"@type\\": \\"WebPage\\",
            \\"@id\\": \\"https://example.com/post/test-slug/\\"
          },
          \\"headline\\": \\"test post summary\\",
          \\"image\\": {
            \\"@type\\": \\"ImageObject\\",
            \\"url\\": \\"https://example.com/images/3.jpg\\",
            \\"height\\": 900,
            \\"width\\": 1600
          },
          \\"datePublished\\": \\"2018-12-24T00:00:00.000Z\\",
          \\"dateModified\\": \\"2018-12-24T00:00:00.000Z\\",
          \\"author\\": {
            \\"@type\\": \\"Person\\",
            \\"name\\": \\"test author\\"
          },
          \\"description\\": \\"test post summary\\",
          \\"publisher\\": {
            \\"@type\\": \\"Organization\\",
            \\"name\\": \\"test site title\\",
            \\"logo\\": {
              \\"@type\\": \\"ImageObject\\",
              \\"url\\": \\"https://example.com/images/amp_publisher_logo.png\\",
              \\"width\\": 530,
              \\"height\\": 60
            }
          }
        }</script>"
`;

exports[`BlogPostHead When hasAlternate is true BlogPostHead render correctly 1`] = `
"<title>test post title</title><link rel=\\"amphtml\\" href=\\"https://amp.example.com\\" data-react-helmet=\\"true\\"><link rel=\\"canonical\\" href=\\"https://example.com\\" data-react-helmet=\\"true\\"><link rel=\\"alternate\\" href=\\"https://example.com\\" hreflang=\\"en\\" data-react-helmet=\\"true\\"><link rel=\\"alternate\\" href=\\"https://example.com/post/ja/test-slug/\\" hreflang=\\"ja\\" data-react-helmet=\\"true\\"><meta name=\\"description\\" content=\\"post description\\" data-react-helmet=\\"true\\"><meta property=\\"og:title\\" content=\\"test post title\\" data-react-helmet=\\"true\\"><meta property=\\"og:type\\" content=\\"article\\" data-react-helmet=\\"true\\"><meta property=\\"og:image\\" content=\\"https://example.com/images/3.jpg\\" data-react-helmet=\\"true\\"><meta property=\\"og:url\\" content=\\"https://example.com\\" data-react-helmet=\\"true\\"><meta property=\\"og:description\\" content=\\"test post summary\\" data-react-helmet=\\"true\\"><meta property=\\"og:locale\\" content=\\"en\\" data-react-helmet=\\"true\\"><meta property=\\"og:site_name\\" content=\\"test site title\\" data-react-helmet=\\"true\\"><meta property=\\"article:published_time\\" content=\\"2018-12-24T00:00:00.000Z\\" data-react-helmet=\\"true\\"><meta property=\\"profile:username\\" content=\\"test author\\" data-react-helmet=\\"true\\"><meta property=\\"article:tag\\" content=\\"test1\\" data-react-helmet=\\"true\\"><meta property=\\"article:tag\\" content=\\"test2\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:card\\" content=\\"summary_large_image\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:creator\\" content=\\"@testAccount\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:title\\" content=\\"test post title\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:description\\" content=\\"test post summary\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:image\\" content=\\"https://example.com/images/3.jpg\\" data-react-helmet=\\"true\\"><script type=\\"application/ld+json\\" data-react-helmet=\\"true\\">{
        \\"@context\\": \\"http://schema.org\\",
        \\"@type\\": \\"BreadcrumbList\\",
        \\"itemListElement\\": [
          {
            \\"@type\\": \\"ListItem\\",
            \\"position\\": 1,
            \\"item\\": {
              \\"@id\\": \\"https://example.com/\\",
              \\"name\\": \\"HOME\\",
              \\"image\\": \\"https://example.com/images/home.jpg\\"
            }
          },
          {
            \\"@type\\": \\"ListItem\\",
            \\"position\\": 2,
            \\"item\\": {
              \\"@id\\": \\"https://example.com/post/test-slug/\\",
              \\"name\\": \\"test post title\\",
              \\"image\\": \\"https://example.com/images/3.jpg\\"
            }
          }
        ]
      }</script><script type=\\"application/ld+json\\" data-react-helmet=\\"true\\">{
          \\"@context\\": \\"http://schema.org\\",
          \\"@type\\": \\"Article\\",
          \\"mainEntityOfPage\\": {
            \\"@type\\": \\"WebPage\\",
            \\"@id\\": \\"https://example.com/post/test-slug/\\"
          },
          \\"headline\\": \\"test post summary\\",
          \\"image\\": {
            \\"@type\\": \\"ImageObject\\",
            \\"url\\": \\"https://example.com/images/3.jpg\\",
            \\"height\\": 900,
            \\"width\\": 1600
          },
          \\"datePublished\\": \\"2018-12-24T00:00:00.000Z\\",
          \\"dateModified\\": \\"2018-12-24T00:00:00.000Z\\",
          \\"author\\": {
            \\"@type\\": \\"Person\\",
            \\"name\\": \\"test author\\"
          },
          \\"description\\": \\"test post summary\\",
          \\"publisher\\": {
            \\"@type\\": \\"Organization\\",
            \\"name\\": \\"test site title\\",
            \\"logo\\": {
              \\"@type\\": \\"ImageObject\\",
              \\"url\\": \\"https://example.com/images/amp_publisher_logo.png\\",
              \\"width\\": 530,
              \\"height\\": 60
            }
          }
        }</script>"
`;
